Remotion/Web/Core/Services/WebServiceFactory.cs

```diff
@@ -98,11 +100,31 @@ private Type GetAndCheckCompiledType<T> (string virtualPath)
 
     private static IReadOnlyCollection<Tuple<string, IReadOnlyCollection<string>>> GetServiceMethods (Type type)
     {
-      return Array.AsReadOnly (
-          type.GetMethods().Select (
-              mi => Tuple.Create<string, IReadOnlyCollection<string>> (
-                  mi.Name,
-                  Array.AsReadOnly (mi.GetParameters().Select (pi => pi.Name).ToArray()))).ToArray());
+      var visitedTypes = new HashSet<Type>();
+      var typesToVisit = new Queue<Type>();
+      typesToVisit.Enqueue (type);
+
+      var serviceMethods = new List<Tuple<string, IReadOnlyCollection<string>>>();
+
+      while (typesToVisit.Count > 0)
+      {
+        var nextType = typesToVisit.Dequeue();
+        if (!visitedTypes.Add (nextType))
```

Comment:
from a typical-code perspective, this would usually be written with an inverted condition. that way, there's both no `not` and no `continue` *) in the code. I know that it would remove the "it's an guard condition", but this might not be relvant when the rest of the loop code becomes simpler to read. 

*) we try to avoid break and contiune in loops so long as the result code isn't unreadable because we don't have to think about the exit condition while reading the loop.

=> so I'd like to see how the other version looks like before going with continue.